dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/DashbuilderMigrationService.java

```diff
@@ -0,0 +1,154 @@
+import java.util.function.Predicate;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.commons.services.cdi.Startup;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.file.Path;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.file.SimpleFileVisitor;
+import org.uberfire.java.nio.file.attribute.BasicFileAttributes;
+import org.uberfire.java.nio.file.FileVisitResult;
+import org.uberfire.java.nio.file.Files;
+import org.uberfire.java.nio.IOException;
+
+import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;
+
+@ApplicationScoped
+@Startup
+public class DashbuilderMigrationService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DashbuilderMigrationService.class);
+
+    private IOService ioService;
+    private FileSystem oldDatasetsFS;
+    private FileSystem datasetsFS;
+    private FileSystem pluginsFS;
+    private FileSystem perspectivesFS;
+    private FileSystem navigationFS;
+
+    public DashbuilderMigrationService() {
+
+    }
+
+    @Inject
+    public DashbuilderMigrationService(
+            final @Named("ioStrategy") IOService ioService,
+            final @Named("datasetsFS") FileSystem datasetsFS,
+            final @Named("pluginsFS") FileSystem pluginsFS,
+            final @Named("perspectivesFS") FileSystem perspectivesFS,
+            final @Named("navigationFS") FileSystem navigationFS) {
+
+        this.ioService = ioService;
+        this.oldDatasetsFS = ioService.getOrNewFileSystem(URI.create("git://system/datasets"),
+                                                          new HashMap<String, Object>());
+        this.datasetsFS = datasetsFS;
+        this.pluginsFS = pluginsFS;
+        this.perspectivesFS = perspectivesFS;
+        this.navigationFS = navigationFS;
+    }
+
+    @PostConstruct
+    private void init() {
+        migrateDatasets(oldDatasetsFS, datasetsFS);
+        migratePerspectives(pluginsFS, perspectivesFS);
+        migrateNavigation(pluginsFS, navigationFS);
+    }
+
+    public void migrateDatasets(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> true);
+    }
+
+    public void migratePerspectives(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> path.getFileName().toString().startsWith("perspective_layout"));
+    }
+
+    public void migrateNavigation(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> path.getFileName().toString().equals("navtree.json"));
+    }
+
+    private void migrate(FileSystem sourceFS, FileSystem targetFS, Predicate<Path> predicate) {
+        LOGGER.info("Start");
```

Comment:
Could you please be more clear about this message? Like:  Starting filesystem sourceDs  migration to targetDs

Reply:
Done